### PR TITLE
Account settings

### LINKED ISSUE #1199: Implement Account settings

### PR DESCRIPTION
**Pull request by na3d**
Closes #1199 
screen shot 2017-03-10 at 7 19 33 pm

### CODE REVIEW

File: src/api/account.js [hunk 1]
```diff
@@ -14,3 +14,21 @@ function profileAction(action, body, handleRsp) {
...
+    };
+  },
+  put: function(body) {
+    return async (dispatch, getState) => {
+      const state = getState();
+      const { token } = state.authentication;
+      await fetch(token, '/account/settings', { method: 'PUT', body });
+    };
+  },
+};*/
```
Comment (eatonphil) [inline src/api/account.js]: Can we delete?

File: src/settings/layouts/IndexPage.js [hunk 2]
```diff
@@ -3,26 +3,104 @@ import { connect } from 'react-redux';
...
+          <div className="container">
+            <h1>
+              Settings
+            </h1>
+          </div>
+        </header>
+        <div className="container">
+          <Card>
+            <Form onSubmit={() => this.onSubmit()}>
+              <label className="col-sm-2 col-form-label">Enable Network Helper</label>
```
Comment (eatonphil) [inline src/settings/layouts/IndexPage.js]: I think this should be `row-label`? @mparke

**Comment by eatonphil** [comment]
Looks good other than requested changes :+1

File: src/settings/layouts/IndexPage.js (hunk 2 above)
Comment (na3d) [inline src/settings/layouts/IndexPage.js]: `row-label` screws it up, making it text wrap.

Comment (eatonphil) [inline src/settings/layouts/IndexPage.js]: It is probably fine to disable the text wrap for all row-label... especially if col-form-label does

**Comment by eatonphil** [comment]
One last thing, could you add a paragraph saying something like:

These settings apply to all users under this account.

At the top of the card?

**Comment by na3d** [comment]
This is the best I could get it to look:
screen shot 2017-03-15 at 11 11 43 am

**Comment by eatonphil** [comment]
Oh no, I'd put it at the very top above any (of the currently only one) actual settings options.

**Comment by na3d** [comment]
This is how it looks when I do that:
screen shot 2017-03-15 at 11 15 39 am

**Comment by na3d** [comment]
screen shot 2017-03-15 at 11 20 29 am

**Comment by na3d** [comment]
screen shot 2017-03-15 at 11 29 37 am

File: src/components/cards/Card.js [hunk 3]
```diff
@@ -5,11 +5,11 @@ import HelpButton from '../HelpButton';
 export default function Card(props) {
   return (
     <div className={`Card ${props.className}`} id={props.id}>
-      <header className="Card-header clearfix">
+      {props.title !== undefined ? <header className="Card-header clearfix">
```
Comment (mparke) [inline src/components/cards/Card.js]: `props.title ?`

File: src/settings/layouts/IndexPage.js [hunk 4]
```diff
@@ -3,26 +3,109 @@ import { connect } from 'react-redux';
...
+                <label className="col-sm-3 row-label">Enable Network Helper</label>
+                <div className="col-sm-9">
+                  <Checkbox
+                    id="networkHelper"
+                    checked={networkHelper}
+                    onChange={() => this.setState({
+                      networkHelper: !networkHelper,
+                    })}
+                    label="Network Helper automatically deposits a static
+                      networking configuration in to your Linode at boot."
```
Comment (mparke) [inline src/settings/layouts/IndexPage.js]: typo? `in to` > `into`

@pbzona any suggestions on wording here?

File: src/settings/layouts/IndexPage.js [hunk 5]
```diff
@@ -3,26 +3,109 @@ import { connect } from 'react-redux';
...
-  dispatch: PropTypes.func,
+  dispatch: PropTypes.func.isRequired,
+  settings: PropTypes.object.isRequired,
 };
 
-function select() {
-  return {};
+function select(state) {
+  return {
+    settings: state.api.settings.settings.undefined,
```
Comment (mparke) [inline src/settings/layouts/IndexPage.js]: damn, another one of these?

File: src/settings/layouts/IndexPage.js (hunk 4 above)
Comment (pbzona) [inline src/settings/layouts/IndexPage.js]: @mparke I'd go with `into`

File: src/settings/layouts/IndexPage.js (hunk 5 above)
Comment (eatonphil) [inline src/settings/layouts/IndexPage.js]: Didn't see these. I would not use apiActionReducer stuff for these settings (or anywhere that creates these undefineds). I would just write an adhoc getter/setter for now.

Comment (na3d) [inline src/settings/layouts/IndexPage.js]: We are going to run into these until we actually fix the problem.  It is pretty easy to find all of the places we have this problem if we keep consistent with how we handle it.

**Pull request merged**